extensions/wikia/ArticleVideo/bower_components/html5-skin/js/controller.js

```diff
@@ -0,0 +1,1496 @@
+          this.mb.subscribe(OO.EVENTS.SHOW_AD_CONTROLS, "customerUi", _.bind(this.onShowAdControls, this));
+          this.mb.subscribe(OO.EVENTS.SHOW_AD_MARQUEE, "customerUi", _.bind(this.onShowAdMarquee, this));
+        }
+      }
+      this.state.isSubscribed = true;
+    },
+
+    externalPluginSubscription: function() {
+      if (OO.EVENTS.DISCOVERY_API) {
+        this.mb.subscribe(OO.EVENTS.DISCOVERY_API.RELATED_VIDEOS_FETCHED, "customerUi", _.bind(this.onRelatedVideosFetched, this));
+      }
+    },
+
+    /*--------------------------------------------------------------------
+     event listeners from core player -> regulate skin STATE
+     ---------------------------------------------------------------------*/
+    onPlayerCreated: function (event, elementId, params, settings) {
+      //set state variables
+      this.state.mainVideoContainer = $("#" + elementId);
+      this.state.mainVideoInnerWrapper = $("#" + elementId + " .innerWrapper");
+      this.state.playerParam = params;
+      this.state.elementId = elementId;
+      this.state.isMobile = Utils.isMobile();
+      this.state.browserSupportsTouch = Utils.browserSupportsTouch();
+
+      //initial DOM manipulation
+      this.state.mainVideoContainer.addClass('oo-player-container');
+      this.state.mainVideoInnerWrapper.addClass('oo-player');
+      this.state.mainVideoInnerWrapper.append("<div class='oo-player-skin'></div>");
+
+      //load player with page level config param if exist
+      if (params.skin && params.skin.config) {
+        $.getJSON(params.skin.config, function(data) {
+          this.loadConfigData(params, settings, data);
+        }.bind(this));
+      }
+      //else load player with bundled config
+      else {
+        this.loadConfigData(params, settings);
+      }
+
+      var accessibilityControls = new AccessibilityControls(this); //keyboard support
+      this.state.screenToShow = CONSTANTS.SCREEN.LOADING_SCREEN;
+    },
+
+    onVcVideoElementCreated: function(event, params) {
+      var element = $("#" + params["domId"]);
+      var elementVideo = element.find("video");
+
+      //if video element is descendant
+      if (elementVideo.length) {
+        element = elementVideo;
+        this.state.mainVideoMediaType = CONSTANTS.MEDIA_TYPE.HTML5;
+      }
+
+      //add loadedmetadata event listener to main video element
+      if (element.get(0)) {
+        element.get(0).addEventListener("loadedmetadata", this.metaDataLoaded.bind(this));
+      }
+
+      if (Utils.isIE10()) {
+        element.attr("controls", "controls");
+      }
+
+      if (params["videoId"] === OO.VIDEO.MAIN) {
+        this.state.mainVideoElement = element;
+        this.enableFullScreen();
+        this.updateAspectRatio();
+      }
+    },
+
+    // functions dependent on video metadata
+    metaDataLoaded: function () {
+      this.enableIosFullScreen();
+    },
+
+    onPlayerDestroy: function (event) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove the unused function parameter "event". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)